api/src/main/java/org/pmiops/workbench/cohortbuilder/CohortBuilderServiceImpl.java

```diff
@@ -312,6 +317,84 @@ public CriteriaListWithCountResponse findCriteriaByDomain(
+    String modifiedSearchTerm = parsedTerms.get(MODIFIED_TERMS);
+    List<String> endsWithTerms = new ArrayList<>();
+    if (parsedTerms.get(ENDS_WITH_TERMS).length() > 0) {
+      endsWithTerms.addAll(Arrays.asList(parsedTerms.get(ENDS_WITH_TERMS).split(",")));
+    }
+
+    // if the modified search term is empty and endsWithTerms is empty return an empty result
+    if (modifiedSearchTerm.length() == 0 && endsWithTerms.isEmpty()) {
+      return new CriteriaListWithCountResponse().totalCount(0L);
+    }
+
+    // if domain type is survey and modifiedSearchTerm is not empty then return search survey
+    if (isSurveyDomain(domain) && modifiedSearchTerm.length() > 0) {
+      return findSurveyCriteriaBySearchTerm(surveyName, pageRequest, modifiedSearchTerm);
+    }
+
+    List<DbCriteria> dbCriteriaList = new ArrayList<>();
+    // find a match on concept code
+    dbCriteriaList.addAll(
+        cbCriteriaDao
+            .findCriteriaByDomainAndTypeAndCodeAndStandard(
+                domain, term.replaceAll("[()+\"*-]", ""), standard, pageRequest)
+            .getContent());
+
+    // if no match is found on concept code then find match on full text index by term
+    if (dbCriteriaList.isEmpty() && !term.contains(".")) {
+      if (endsWithTerms.isEmpty()) {
+        // regular search with modified terms, no endsWithTerms
+        dbCriteriaList.addAll(
+            cbCriteriaDao
+                .findCriteriaByDomainAndFullTextAndStandard(
+                    domain, modifiedSearchTerm, standard, pageRequest)
+                .getContent());
+      } else if (!endsWithTerms.isEmpty() && modifiedSearchTerm.length() == 0) {
+        // search only endsWithTerms for each endsWithTerm
+        endsWithTerms.stream()
+            .forEach(
+                endsTerm -> {
+                  dbCriteriaList.addAll(
+                      cbCriteriaDao
+                          .findCriteriaByDomainAndStandardAndNameEndsWith(
+                              domain, standard, endsTerm, pageRequest)
+                          .getContent());
+                });
+      } else if (!endsWithTerms.isEmpty() && modifiedSearchTerm.length() > 0) {
+        // search for eachEndsWithTerm and modifiedSearchTerm
+        endsWithTerms.stream()
+            .forEach(
+                endsTerm -> {
+                  dbCriteriaList.addAll(
+                      cbCriteriaDao
+                          .findCriteriaByDomainAndStandardAndTermAndNameEndsWith(
+                              domain, standard, modifiedSearchTerm, endsTerm, pageRequest)
+                          .getContent());
+                });
+      }
+    }
+
+    return new CriteriaListWithCountResponse()
+        .items(
+            dbCriteriaList.stream()
+                .map(cohortBuilderMapper::dbModelToClient)
+                .collect(Collectors.toList()))
+        .totalCount(Long.valueOf(dbCriteriaList.size()));
```

Comment:
This needs to be the total results count, not the size of the pagination.

Reply:
addressed this as per discussion